Screen Wake Lock API subedit (#31208)
mdn/content: files/en-us/web/api/wakelocksentinel/index.md

{{securecontext_header}}{{APIRef("Screen Wake Lock API")}}

> The **`WakeLockSentinel`** interface of the [Screen Wake Lock API](/en-US/docs/Web/API/Screen_Wake_Lock_API) can be used to monitor the status of the platform screen wake lock, and manually release the lock when needed.

Author: This is probably the only necessary fix in the doc as it implies that the sentinel can be used to reacquire the lock. That is not the case - it is single use, though it does tell you when the lock is lost and may need to be reacquired.

The screen wake lock prevents device screens from dimming or locking when an application needs to keep running.

> A screen wake lock is requested using the {{domxref('WakeLock.request()','navigator.wakeLock.request()')}} method, which returns a {{jsxref('Promise')}} that fulfills to an object of this type if the lock is granted.

Reviewer: ```suggestion
A screen wake lock is requested using the {{domxref('WakeLock.request()','navigator.wakeLock.request()')}} method, which returns a {{jsxref('Promise')}} that fulfills with a `WakeLockSentinel` object if the lock is granted.
```

Reviewer: Just so it is completely obvious what is happening.

> An acquired screen wake lock can be released manually via the {{domxref('WakeLockSentinel.release','release()')}} method, or automatically via the platform screen wake lock (this might happen if the document becomes inactive or looses visibility, if the device is low on power, or the user turns on a power save mode).

Reviewer: ```suggestion
An acquired screen wake lock can be released manually via the {{domxref('WakeLockSentinel.release','release()')}} method, or automatically via the platform screen wake lock. The latter may occur if the document becomes inactive or loses visibility, if the device is low on power, or if the user turns on a power save mode.
```

## Examples

> In this example we create an asynchronous function which requests a `WakeLockSentinel`.

Reviewer: ```suggestion
In this example, we create an asynchronous function that requests a `WakeLockSentinel`.
```

> Once acquired we listen for the `release` event which can be used to give appropriate UI feedback.

Reviewer: ```suggestion
Once the screen wake lock is acquired we listen for the `release` event, which can be used to give appropriate UI feedback.
```
